addons/overpressure/functions/fnc_firedEHOP.sqf

```diff
@@ -22,7 +22,10 @@ TRACE_8("firedEH:",_unit,_weapon,_muzzle,_mode,_ammo,_magazine,_projectile,_gunn
 private _opValues = [_weapon, _ammo, _magazine] call FUNC(getOverPressureValues);
 
 _opValues params ["_dangerZoneAngle", "_dangerZoneRange", "_dangerZoneDamage"];
-_dangerZoneRange = _dangerZoneRange * GVAR(overpressureDistanceCoefficient);
+
+private _unitOverpressureRangeAttribute = _unit getVariable [QEGVAR(overpressure,distance), GVAR(overpressureDistanceCoefficient)];
+
+_dangerZoneRange = _dangerZoneRange * ([_unitOverpressureRangeAttribute, GVAR(overpressureDistanceCoefficient)] select (_unitOverpressureRangeAttribute < 0));
 
 TRACE_3("cache",_dangerZoneAngle,_dangerZoneRange,_dangerZoneDamage);
```

Comment:
```
TRACE_3("cache",_dangerZoneAngle,_dangerZoneRange,_dangerZoneDamage);

if (_dangerZoneDamage <= 0) exitWith {};

private _unitOverpressureRangeAttribute = _unit getVariable [QGVAR(distance), GVAR(overpressureDistanceCoefficient)];
_dangerZoneRange = _dangerZoneRange * _unitOverpressureRangeAttribute;

// The weapon produces overpressure, calculate
```
moving the distance calcs bellow the damage exitWith is a nice small optimization
I can't do this in a suggestion but we can always just merge as-is and fix in a followup